frontend/src/modules/EditZing/Components/EditZing.tsx

```diff
@@ -0,0 +1,188 @@
+  const [hasLoadedStudentData, setHasLoadedStudentData] = useState(false)
+  useEffect(() => {
+    axios
+      .get(`${API_ROOT}${COURSE_API}/students/${courseId}`)
+      .then((res: AxiosResponse<CourseStudentDataResponse>) => {
+        setUnmatchedStudents(res.data.data.unmatched)
+        setStudentGroups(res.data.data.groups)
+        setHasLoadedStudentData(true)
+      })
+      .catch((error) => {
+        console.error(error)
+        setShowError(true)
+      })
+  }, [courseId])
+
+  /** Add an unmatched student to a group */
+  const moveStudentFromUnmatched = (
+    student: Student,
+    toGroupNumber: number
+  ) => {
+    setUnmatchedStudents(
+      unmatchedStudents.filter((s) => s.email !== student.email)
+    )
+    setStudentGroups(
+      studentGroups.map((group) =>
+        group.groupNumber === toGroupNumber
+          ? { ...group, memberData: [...group.memberData, student] }
+          : group
+      )
+    )
+    axios
+      .post(`${API_ROOT}${MATCHING_API}/transfer/unmatched`, {
+        courseId: courseId,
+        studentEmail: student.email,
+        groupNumber: toGroupNumber,
+      })
+      .catch((err) => console.error(err))
+  }
+
+  /** Move a student already in a group back to unmatched */
+  const moveStudentToUnmatched = (
+    student: Student,
+    fromGroupNumber: number
+  ) => {
+    setUnmatchedStudents([...unmatchedStudents, student])
+    setStudentGroups(
+      studentGroups.map((group) =>
+        group.groupNumber === fromGroupNumber
+          ? {
+              ...group,
+              memberData: group.memberData.filter(
+                (s) => s.email !== student.email
+              ),
+            }
+          : group
+      )
+    )
+    axios
+      .post(`${API_ROOT}${MATCHING_API}/transfer/unmatch`, {
+        courseId: courseId,
+        studentEmail: student.email,
+        groupNumber: fromGroupNumber,
+      })
+      .catch((err) => console.error(err))
+  }
+
+  /** Transfer a student from a group to another group */
+  const moveStudentIntergroup = (
+    student: Student,
+    fromGroupNumber: number,
+    toGroupNumber: number
+  ) => {
+    setStudentGroups(
+      studentGroups.map((group) =>
+        group.groupNumber === toGroupNumber
+          ? { ...group, memberData: [...group.memberData, student] }
+          : group.groupNumber === fromGroupNumber
+          ? {
+              ...group,
+              memberData: group.memberData.filter(
+                (s) => s.email !== student.email
+              ),
+            }
+          : group
+      )
+    )
+    axios
+      .post(`${API_ROOT}${MATCHING_API}/transfer/intergroup`, {
+        courseId: courseId,
+        studentEmail: student.email,
+        group1: fromGroupNumber,
+        group2: toGroupNumber,
+      })
+      .catch((err) => console.error(err))
+  }
+
+  /** Move a student from some group (existing/unmatched) to a group */
+  const moveStudent = (
```

Comment:
maybe more descriptive function name here

Reply:
sike ignore me i just realized this is a daddy function

Reply:
maybe make more clear in function comment tho

Reply:
Im getting an email for every comment you make lol (maybe try doing them all as a review instead of as a single comment)